cpp/include/cudf/strings/string_view.cuh

```diff
@@ -41,8 +46,14 @@ __device__ inline size_type characters_in_string(const char* str, size_type byte
 {
   if ((str == 0) || (bytes == 0)) return 0;
   auto ptr = reinterpret_cast<uint8_t const*>(str);
+#ifndef CUDF_JIT_UDF
   return thrust::count_if(
     thrust::seq, ptr, ptr + bytes, [](uint8_t chr) { return is_begin_utf8_char(chr); });
+#else
+  size_type chars = 0;
+  while (ptr < ptr + bytes) { chars += is_begin_utf8_char(*ptr++); }
```

Comment:
I don't believe this will work as expected. Per is mutating inside the whole loop, but used as a limit. Maybe `auto end = ptr + bytes` before the while to capture it?

Reply:
Ooops. You are right. Good catch.